stan/math/rev/fun/from_var_value.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef STAN_MATH_REV_FUN_FROM_VAR_VALUE_HPP
+#define STAN_MATH_REV_FUN_FROM_VAR_VALUE_HPP
+
+#include <stan/math/rev/core.hpp>
+#include <stan/math/prim/fun/Eigen.hpp>
+#include <stan/math/rev/functor/arena_matrix.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+
+namespace stan {
+namespace math {
+
+/**
+ * Converts `var_value` into an Eigen Matrix. Adjoint is propagated back to
+ * argument in the reverse pass.
+ *
+ * @tparam T type of the input
+ * @param a matrix to convert
+ */
+template <typename T, require_var_vt<is_eigen, T>* = nullptr>
+Eigen::Matrix<var, T::RowsAtCompileTime, T::ColsAtCompileTime> from_var_value(
+    const T& a) {
+  arena_matrix<Eigen::Matrix<var, T::RowsAtCompileTime, T::ColsAtCompileTime>>
+      res(a.val());
+  reverse_pass_callback([res, a]() mutable { a.vi_->adj_ += res.adj(); });
+  return res;
+}
+
```

Comment:
[optional/questionable] Should we have a version that's a no-op for a `Matrix<var>`?

Reply:
I don't see the need right now. If we ever do need it, it is simple to add.